numpy/core/include/numpy/ndarraytypes.h

```diff
@@ -895,10 +896,14 @@ typedef int (PyArray_FinalizeFunc)(PyArrayObject *, PyObject *);
 #define NPY_ARRAY_OUT_ARRAY    (NPY_ARRAY_CARRAY)
 #define NPY_ARRAY_INOUT_ARRAY  (NPY_ARRAY_CARRAY | \
                                 NPY_ARRAY_UPDATEIFCOPY)
+#define NPY_ARRAY_INOUT_ARRAY2 (NPY_ARRAY_CARRAY | \
+                                NPY_ARRAY_WRITEBACKIFCOPY)
 #define NPY_ARRAY_IN_FARRAY    (NPY_ARRAY_FARRAY_RO)
 #define NPY_ARRAY_OUT_FARRAY   (NPY_ARRAY_FARRAY)
 #define NPY_ARRAY_INOUT_FARRAY (NPY_ARRAY_FARRAY | \
                                 NPY_ARRAY_UPDATEIFCOPY)
+#define NPY_ARRAY_INOUT_FARRAY2 (NPY_ARRAY_FARRAY | \
+                                NPY_ARRAY_WRITEBACKIFCOPY)
```

Comment:
So presumably this `2` version is only needed during the deprecation phase, afterwards we will redefine `NPY_ARRAY_INOUT_FARRAY` to use `NPY_ARRAY_WRITEBACKIFCOPY`.

Reply:
That's one option, but it's a bit risky since we don't have any way to know when all the old code using `NPY_ARRAY_INOUT_FARRAY` is gone. We might just keep the `2` version forever – it's not pretty but it's safe and works.

Reply:
yeah we can leave the 2 version forever, but after the deprecation phase no one needs to use it any more.